src/azure-cli/azure/cli/command_modules/acr/import.py

```diff
@@ -117,11 +112,27 @@ def _handle_result(cmd, result_poller, source_registry, source_image, registry):
                         registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry, source_registry)
 
                 if registry.login_server.lower() in source_image.lower():
-                    logger.warning("Import from source failed.\n\tsource image: '%s'", "{}/{}"
+                    logger.warning("Import from source failed.\n\tsource image: '%s'\n"
+                                   "Attention: When source registry is specified with --registry, "
+                                   "--source is considered to be a source image name. ", "{}/{}"
                                    .format(registry.login_server, source_image))
         except (ClientException, CLIError) as unexpected_ex:  # raise exception
             logger.debug("Unexpected exception: %s", unexpected_ex)
 
         raise e  # regardless re-raise the CLIError as this is an error from the service
 
     return result
+
+
+def _split_registry_and_image(source_image):
+    slash = source_image.find('/')
+    if slash <= 0:
+        raise CLIError(SOURCE_NOT_FOUND)
+
+    registry_uri = source_image[:slash]
+    dot = registry_uri.find('.')
+    if dot <= 0:
+        raise CLIError(LOGIN_SERVER_NOT_VALID)
```

Comment:
Given dot is only used to check for the presence of `.` consider 
```suggestion
    if '.' not in registry_uri:
        raise CLIError(LOGIN_SERVER_NOT_VALID)
```